### ISSUE TITLE
Request: allow to make faction ship/ship class specific outfit

### ISSUE THREAD

**Issue by binaryspica** [endless-sky/endless-sky#7926]
### Problem Description

I want to make the old ships better, but without edit its stat all I could is adding the outfit or making the custom outfits. The problem is, if you make the custom outfits, it can also be used for the better races' ships, despite the item is purely for low tier ships such as Human ships. And the end result is just making the better ship even better, not making the weak ship a chance.

Also there are old problems such as the uselessness of fighter/drone class. So it would be better to also set the ship class as the requirement of the outfit.

### Related Issue Links

https://steamcommunity.com/app/404410/discussions/0/3473990693211643801/

### Desired Solution

Tier is not a golden rule, so 'race/faction specific' could be more accurate. So, what about to set the origin faction as the requirement for the outfit? Such as 'human ship only', 'Hai ship only', and 'Korath Exile ships only'? With the ships' faction can be a requirement for the outfit, then it is possible to only enhance the underdog ships by making some mods.

An another idea is, what about 'ship class specific' option? Such as 'warship only' or 'light warship only', so light warships have some advantage for using such outfits, making some difference with the heavier warships.

### Alternative Approaches

Else it could be just add unique slots for all ships in the faction ships, as Spinal Mount, but it does not allows to add it on the ships we already have.

**Comment by Hurleveur** [comment]
I feel like low tier outfits can keep having a use into the late game by filling niches (like high heat or specific, and what I love about this game precisely is being able to put all kinds of outfits to all kinds of ships.

**Comment by Zireael07** [comment]
Who said anything about doing this for ALL outfits?

**Comment by binaryspica** [comment]
> I feel like low tier outfits can keep having a use into the late game by filling niches (like high heat or specific things they do well or they're the only ones that can do), and what I love about this game precisely is being able to put all kinds of outfits to all kinds of ships. So I'd be against doing this for all outfits.

The request is purely for the **possibility** to make the **mods** that adds faction ship/ship class specific outfit, that is intended to increase the stats of the low tier ships but it must not applied to the higher tier one. The request does **not** ask anything for the basic gameplay or balance at all.

**Comment by Hurleveur** [comment]
Well ofc course more mechanics cant hurt I was just saying that I think thats not the best goal or way to balance things out for vanilla.

**Comment by Hurleveur** [comment]
I do see this could be useful, of course, but also how it is possible to do with some attributes that in the end would not be much more limiting in terms of amounts of lines to write than doing it like we currently can.
As in, make all human ships have "human component" 1000 and all human outfits have "human component" -1 and voila for your presented use case; wouldnt be worse than doing it with a new mechanic?

**Comment by binaryspica** [comment]
> Well ofc course more mechanics cant hurt I was just saying that I think that the goal presented may not the best and I dont believe should be used as a way to balance things out for vanilla.
> 
> Edit: ironically I did do a faction specific bomb but I only gave it to some ships of said faction. I dont know what should be limited to all ships of a specie, maybe give a bit of an example?

I don't think that it should be normal on vanila. I love the hybrid nature of the outfit system, and I don't want most outfits are exclusive for the specific race either. 

I was imagine the additional outfit by mod that makes lesser tier ships as good as the higher tier ships that can be bought by those race, so everything on vanila are need to be kept. So, you are start by vanila by default, but if you encounters a faction and allowed to buy their ships, you could 'upgrade' those ships. It is weird that every single Leviathans(yes, not only yours, but also any human factions have) are durable as much as a Hurricane since the first day you start your first day of captain on New Boston, isn't?

What I was imagined was a mod that available such outfit, that is costs as much credit as the same ship class of the higher tier race and requires the license if you need it to buy their warships, thus the end result in credit spend is not so different. You could make a Leviathan similar with a Hurricane, but in doing this you need both the license to actually buy a Hurricane(and since Hurricane is only available for late part of the quest line so their heavy warship enhancement is only available after you could buy a Hurricane as well) and also need enough credit to buy a Hurricane(or at least the difference), so you have not much a merit over simply buy an another Hurricane instead. Such outfits are need to be exclusive for human ships - and for Hai and Remnant ships, these needs their own exclusive outfits for making them similar to a Hurricane. That's why it needs to be exclusive for some types of ships. For now no one can stop to add such outfits on a Hurricane itself, 349 or 512.

Also that's only the possibility for adding the options. Again, I don't want it to be normal on vanila outfits at all. These must be compatible for every single ships we can have(save for some exception, such as spinal mount, of course). Ship specific extra options are usually only need for the mods, and I am actually against to add such options on the outfits we can see on vanila. Perhaps, Ka'het would be the exception, due to their unique ship structure? But I am not sure.

It is reasonable that add such an option will cause the fear that some outfits we can access are exclusive for some type of ships, though.

An another problem is, is it worthy enough to actually added onto the game?

**Comment by Hurleveur** [comment]
Ships are saved in the savefiles so the mod wouldnt work on old ships whatever we do so long as we dont fix that

**Comment by binaryspica** [comment]
> Ships are saved in the savefiles so the mod wouldnt work on old ships whatever we do so long as we dont fix that

Yes that's the problem as well. So if the feature is added then it should be the requirement on the outfits, for outfits are not saved in the savefiles.

**Comment by Hurleveur** [comment]
You cant just make it a requirement on outfits and not link them to ships
Could do some new structure that would link items together though

**Comment by Amazinite** [comment]
I think this could be achieved with some combination of #11415 and the ship filters mentioned in #11318.
